File: backend/Modules/TableManagement/Entities/DiningTable.cs
using RestaurantPOS.Infrastructure.Common;

namespace RestaurantPOS.Modules.TableManagement.Entities;

public class DiningTable : BaseEntity
{
    public Guid BranchId { get; set; }
    public Guid? ZoneId { get; set; } // Liên kết đến bảng Zones
    public string TableNumber { get; set; } = string.Empty;
    public int Capacity { get; set; }
    public string? QrCode { get; set; }
    public string Status { get; set; } = "Available";

    // Vị trí trên sơ đồ (0-100%)
    public double PosX { get; set; } = 0;
    public double PosY { get; set; } = 0;

    // Navigation
    public Zone? Zone { get; set; }
}

File: mobile_app/lib/models/dining_table.dart
class DiningTable {
  final String id;
  final String tableNumber;
  final String? zoneName;
  final String? branchName;
  final int capacity;
  final String status; // Available, Occupied
  final double posX;
  final double posY;

  DiningTable({
    required this.id,
    required this.tableNumber,
    this.zoneName,
    this.branchName,
    required this.capacity,
    required this.status,
    this.posX = 0,
    this.posY = 0,
  });

  factory DiningTable.fromJson(Map<String, dynamic> json) {
    return DiningTable(
      id: json['id'],
      tableNumber: json['tableNumber'],
      zoneName: json['zoneName'],
      branchName: json['branchName'],
      capacity: json['capacity'] ?? 4,
      status: json['status'] ?? 'Available',
      posX: (json['posX'] ?? 0).toDouble(),
      posY: (json['posY'] ?? 0).toDouble(),
    );
  }
}
